extra_scripts/create_imagenet_a_data_files.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import argparse
+import os
+from typing import Dict
+
+import numpy as np
+import torchvision.datasets as datasets
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+
+def get_argument_parser():
+    """
+    List of arguments supported by the script
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=str,
+        help="Path to the folder containing the original imagenet-a folder",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        help="Folder where the classification dataset will be written",
+    )
+    parser.add_argument(
+        "-d",
+        "--download",
+        action="store_const",
+        const=True,
+        default=False,
+        help="To download the original dataset in the input folder",
+    )
+    return parser
+
+
+def download_dataset(root: str):
+    """
+    Download the Imagenet-R dataset archive and expand it
+    in the folder provided as parameter
+    """
+    URL = "https://people.eecs.berkeley.edu/~hendrycks/imagenet-a.tar"
+    download_and_extract_archive(url=URL, download_root=root)
+
+
+def create_imagenet_test_files(
+    input_path: str, output_path: str, target_mappings: Dict[int, int]
+):
+    """
+    Create a test split for ImageNet, by reading the image folder in the
+    'input_path' and mapping its targets to the targets of imagenet
+    using the 'target_mappings' dictionary
+    """
+
+    # Map the images of the image folder to their corresponding targets
+    # in ImageNet
+    image_paths = []
+    image_labels = []
+    imagenet_r = datasets.ImageFolder(root=input_path, loader=lambda x: x)
+    for image_path, target_id in tqdm(imagenet_r):
+        image_paths.append(image_path)
+        image_labels.append(target_mappings[target_id])
+
+    # Save the these lists in the disk_filelist format
+    os.makedirs(output_path, exist_ok=True)
+    img_info_out_path = os.path.join(output_path, "test_images.npy")
+    label_info_out_path = os.path.join(output_path, "test_labels.npy")
+    np.save(img_info_out_path, np.array(image_paths))
+    np.save(label_info_out_path, np.array(image_labels))
```

Comment:
Can you look into using the `PathManager` here? https://github.com/facebookresearch/vissl/blob/master/extra_scripts/create_voc_data_files.py#L16